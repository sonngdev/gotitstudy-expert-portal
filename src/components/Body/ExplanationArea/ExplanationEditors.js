import React from 'react';
import { TinyEditor } from './TinyEditor';
import './styles/ExplanationEditors.css';

export function ExplanationEditors() {
  return (
    <div className="explanation-editors u-paddingRightSmall">
      <h5>Theory or Concept</h5>
      <TinyEditor id="concept-editor" />

      <h5 className="u-marginTopLarge">Step By Step Instructions</h5>
      <TinyEditor id="instructions-editor" />

      <h5 className="u-marginTopLarge">Final Answer</h5>
      <TinyEditor id="answer-editor" />
    </div>
  );
}
