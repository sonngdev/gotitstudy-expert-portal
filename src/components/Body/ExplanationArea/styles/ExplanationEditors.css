.explanation-editors {
  height: calc(100vh - 235px);
  overflow: auto;
}
