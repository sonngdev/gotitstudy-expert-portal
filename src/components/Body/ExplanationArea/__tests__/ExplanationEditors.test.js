import React from 'react';
import { shallow } from 'enzyme';
import { ExplanationEditors } from '../ExplanationEditors';

describe('<ExplanationEditors />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ExplanationEditors />);
  });

  it('renders', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('contains three sections, each with title and editor', () => {
    expect(wrapper.find('h5')).toHaveLength(3);
    expect(wrapper.find('TinyEditor')).toHaveLength(3);
  });

  it('has Theory or Concept as the first section', () => {
    expect(wrapper.find('h5').at(0).text()).toBe('Theory or Concept');
  });

  it('has Step By Step Instructions as the second section', () => {
    expect(wrapper.find('h5').at(1).text()).toBe('Step By Step Instructions');
  });

  it('has Final Answer as the third section', () => {
    expect(wrapper.find('h5').at(2).text()).toBe('Final Answer');
  });
});
