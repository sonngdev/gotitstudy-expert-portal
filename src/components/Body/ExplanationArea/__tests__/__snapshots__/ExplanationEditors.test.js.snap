// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExplanationEditors /> renders 1`] = `
"<div className=\\"explanation-editors u-paddingRightSmall\\">
  <h5>
    Theory or Concept
  </h5>
  <TinyEditor id=\\"concept-editor\\" onChange={[Function: onChange]} />
  <h5 className=\\"u-marginTopLarge\\">
    Step By Step Instructions
  </h5>
  <TinyEditor id=\\"instructions-editor\\" onChange={[Function: onChange]} />
  <h5 className=\\"u-marginTopLarge\\">
    Final Answer
  </h5>
  <TinyEditor id=\\"answer-editor\\" onChange={[Function: onChange]} />
</div>"
`;
